doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,352 @@
+    flow.
+  - Control various specific actuators such as motors, solenoids, or valves.
+  - Collect and send data with a low latency, or at least a well synchronized
+    timestamp.
+  - Potentially have a very low unit cost, particularly when the networks are
+    implemented in mass market products such as automobiles.
+  - Potentially run control algorithms themselves.
+
+## 2.5. Edge systems for building management and automation
+
+Various sensing and environmental control tasks can be done by edge nodes in
+buildings. The nodes need to be able to:
+
+  - Sense temperature, light, humidity, pressure, current, etc.
+  - Control temperature, light, ventilation, etc.
+  - Connect to the building management system, via wired or wireless connectors.
+  - Deliver data with controllable timing and accuracy.
+
+## 2.6. Smart home devices
+
+Easy to use devices can let consumers monitor and control their home
+automatically. The devices need to be able to:
+
+  - Sense temperature, light, humidity, smoke, etc.
+  - Integrate with home appliances to control temperature, light, power usage,
+    etc.
+  - Be usable but secure for people with no technical knowledge.
+  - Connect to a commercial home gateway.
+  - Communicate over common home IoT protocols such as ZigBee, Bluetooth, Wi-Fi,
+    etc.
+  - Connect to the building's power or provide their own power, depending on
+    the product.
+
+## 2.7. Daughterboards
+
+Plug-in boards can give devices immediate support for a protocol or standard,
+or let them outsource a task from the main processor. This requires the board
+to:
+
+  - Perform its promised task reliably.
+  - Integrate and communicate reliably with the main processor.
+  - Include and support any relevant hardware.
+
+## 2.8. Education
+
+The broad technical scope of RIOT makes it useful as a basis for education.
+This requires:
+
+  - The presence of didactic materials related to RIOT.
+  - Clear and easy to find documentation.
+  - A general availability of supported hardware and tools.
+
+# 3. Design philosophies
+
+Below are the design philosophies that are typically followed by developers to
+cater for the above use cases. The sections below include descriptions of
+tradeoffs between the philosophies, and where the resolutions typically fall.
+
+## Suitability for constrained devices
+
+"Constrained" means that available memory, energy, and processor cycles are so
+reduced as to become a dominant consideration in design requirements [1]. 
+
+#### Energy efficiency
+
+RIOT nodes sometimes need to last for several years without external power, so
+they need to manage energy carefully. RIOT's tickless scheduler lets devices
+sleep while they aren't collecting data.
+
+Modules outside the core should leverage the benefits and address the
+programming challenges of such a scheduler. They shouldn't demand that users do
+the same.  They should, however, allow users to manage power through different
+modes and functions.
+
```

Comment:
These points fit more in the "design philosophies" section as they are more about internal design of RIOT rather than the use cases.  
Some aspects you've mentioned are in there already, but what currently isn't is differentiation between energy management which is "magic" (invisible) and that which is user configurable. Just to clarify, are you suggesting the following: we balance usability with energy efficiency, except on some selected platforms (Hamilton? etc) where we do indeed squeeze the last Joule out in the default configuration?

Reply:
(see the section titled "energy efficiency")

Reply:
Clarified the section about energy efficiency offline, with some initial thoughts. @MichelRottleuthner will think more about any further clarifications.